### DISCUSSION TITLE
curl: (7) Failed to connect to localhost port 4000 after 2234 ms.

### DISCUSSION THREAD

**Discussion by Denel91** [lifeisfoo/react-guida-pratica#6] [Q&A] [answered]
I have a problem to run the localhost:4000 
The CMD prompt show this message: 
curl: (7) Failed to connect to localhost port 4000 after 2234 ms.

Curl is correctly installed how you can see...
C:\Users\utente>curl -V
curl 7.83.1 (Windows) libcurl/7.83.1 Schannel
Release-Date: 2022-05-13
Protocols: dict file ftp ftps http https imap imaps pop3 pop3s smtp smtps telnet tftp
Features: AsynchDNS HSTS IPv6 Kerberos Largefile NTLM SPNEGO SSL SSPI UnixSockets

Someone can help me?

**Comment by lifeisfoo** [comment]
Ciao @Denel91, essendo il supporto per il libro in italiano, ti rispondo in italiano, spero non sia un problema.

Potresti spiegare meglio quale problema hai di preciso? Dai log che hai incollato non è chiaro.

Non riesci ad eseguire il server delle API? Lo esegui ma non riesci a raggiungerlo con `curl`?

**Reply by Denel91** [reply]
Buonasera,
Praticamente il comando curl localhost:4000/api/lists fallisce. Ho verificato quindi di avere il comando curl a sistema e sembrerebbe di sì infatti mi da la versione corrente. 
Vorrei chiederle se c'è un altro modo per poter accedere alle api, per esempio mettendo la cartella api dentro il progetto. 

Grazie Mille.

**Reply by lifeisfoo** [reply]
Mi potresti scrivere il comando che hai usato per lanciare il server API? Potresti anche incollarmi il log prodotto dal server quando viene eseguito? Grazie

**Comment by Denel91** [comment]
utente@DESKTOP-G4O0D0I MINGW64 ~/Desktop (master)
$ curl localhost:4000/api/lists
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:--  0:00:02 --:--:--     0
curl: (7) Failed to connect to localhost port 4000 after 2250 ms: Couldn't connect to server

**Reply by lifeisfoo** [reply, marked as answer]
Questo è il comando per richiedere le liste al server. 
Segui le istruzioni del Capitolo 10, nel paragrafo "API per le attività".
Se non riesci a lanciare il server seguendo quelle istruzioni, incolla qui l'errore che ricevi.

In generale, prima di usare curl, devi assicurarti che il server delle API sia in esecuzione (p. 157).

**Reply by Denel91** [reply]
Ci sono riuscito...Avevo io un problema con Node.js e le variabili d'ambiente di sistema.
Funziona!!!
Grazie Mille.
